netlib/http2/protocol.py

```diff
@@ -48,16 +49,19 @@ def check_alpn(self):
                 "HTTP2Protocol can not handle unknown ALP: %s" % alp)
         return True
 
-    def _receive_settings(self):
-        frm = frame.Frame.from_file(self.tcp_handler.rfile, self)
-        assert isinstance(frm, frame.SettingsFrame)
-        self._apply_settings(frm.settings)
+    def _receive_settings(self, hide=False):
+        while True:
+            frm = self.read_frame(hide)
+            if isinstance(frm, frame.SettingsFrame):
+                break
```

Comment:
This function looks weird. We just read frames and discard them until we stumble upon a settings frame? Don't we swallow possibly relevant packets by this?